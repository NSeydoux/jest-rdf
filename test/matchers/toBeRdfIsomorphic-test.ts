import {blankNode, namedNode, quad} from "@rdfjs/data-model";
import "../../index";

describe('#toBeRdfIsomorphic', () => {
  it('should succeed for equal quad arrays', () => {
    return expect([
      quad(
        namedNode('s1'),
        namedNode('p1'),
        namedNode('o1'),
        namedNode('g1'),
      ),
      quad(
        namedNode('s2'),
        namedNode('p2'),
        namedNode('o2'),
        namedNode('g2'),
      ),
      quad(
        namedNode('s3'),
        namedNode('p3'),
        namedNode('o3'),
        namedNode('g3'),
      ),
    ]).toBeRdfIsomorphic([
      quad(
        namedNode('s1'),
        namedNode('p1'),
        namedNode('o1'),
        namedNode('g1'),
      ),
      quad(
        namedNode('s2'),
        namedNode('p2'),
        namedNode('o2'),
        namedNode('g2'),
      ),
      quad(
        namedNode('s3'),
        namedNode('p3'),
        namedNode('o3'),
        namedNode('g3'),
      ),
    ]);
  });

  it('should succeed for isomorphic quad arrays', () => {
    return expect([
      quad(
        namedNode('s1'),
        namedNode('p1'),
        blankNode('b1'),
        namedNode('g1'),
      ),
      quad(
        blankNode('b1'),
        namedNode('p2'),
        namedNode('o2'),
        namedNode('g2'),
      ),
      quad(
        blankNode('b1'),
        namedNode('p3'),
        namedNode('o3'),
        namedNode('g3'),
      ),
    ]).toBeRdfIsomorphic([
      quad(
        blankNode('b2'),
        namedNode('p2'),
        namedNode('o2'),
        namedNode('g2'),
      ),
      quad(
        blankNode('b2'),
        namedNode('p3'),
        namedNode('o3'),
        namedNode('g3'),
      ),
      quad(
        namedNode('s1'),
        namedNode('p1'),
        blankNode('b2'),
        namedNode('g1'),
      ),
    ]);
  });

  it('should not fail for equal quad arrays', () => {
    return expect(() => expect([
      quad(
        namedNode('s1'),
        namedNode('p1'),
        namedNode('o1'),
        namedNode('g1'),
      ),
      quad(
        namedNode('s2'),
        namedNode('p2'),
        namedNode('o2'),
        namedNode('g2'),
      ),
      quad(
        namedNode('s3'),
        namedNode('p3'),
        namedNode('o3'),
        namedNode('g3'),
      ),
    ]).not.toBeRdfIsomorphic([
      quad(
        namedNode('s1'),
        namedNode('p1'),
        namedNode('o1'),
        namedNode('g1'),
      ),
      quad(
        namedNode('s2'),
        namedNode('p2'),
        namedNode('o2'),
        namedNode('g2'),
      ),
      quad(
        namedNode('s3'),
        namedNode('p3'),
        namedNode('o3'),
        namedNode('g3'),
      ),
    ])).toThrowErrorMatchingSnapshot();
  });

  it('should not succeed for quad arrays with different length', () => {
    return expect([
      quad(
        namedNode('s1'),
        namedNode('p1'),
        namedNode('o1'),
        namedNode('g1'),
      ),
      quad(
        namedNode('s2'),
        namedNode('p2'),
        namedNode('o2'),
        namedNode('g2'),
      ),
      quad(
        namedNode('s3'),
        namedNode('p3'),
        namedNode('o3'),
        namedNode('g3'),
      ),
    ]).not.toBeRdfIsomorphic([
      quad(
        namedNode('s1'),
        namedNode('p1'),
        namedNode('o1'),
        namedNode('g1'),
      ),
    ]);
  });

  it('should fail for quad arrays with different length', () => {
    return expect(() => expect([
      quad(
        namedNode('s1'),
        namedNode('p1'),
        namedNode('o1'),
        namedNode('g1'),
      ),
      quad(
        namedNode('s2'),
        namedNode('p2'),
        namedNode('o2'),
        namedNode('g2'),
      ),
      quad(
        namedNode('s3'),
        namedNode('p3'),
        namedNode('o3'),
        namedNode('g3'),
      ),
    ]).toBeRdfIsomorphic([
      quad(
        namedNode('s1'),
        namedNode('p1'),
        namedNode('o1'),
        namedNode('g1'),
      ),
    ])).toThrowErrorMatchingSnapshot();
  });

  it('should not succeed for quad arrays with equal length but different contents', () => {
    return expect([
      quad(
        namedNode('s1'),
        namedNode('p1'),
        namedNode('o1'),
        namedNode('g1'),
      ),
      quad(
        namedNode('s2'),
        namedNode('p2'),
        namedNode('o2'),
        namedNode('g2'),
      ),
      quad(
        namedNode('s3'),
        namedNode('p3'),
        namedNode('o3'),
        namedNode('g3'),
      ),
    ]).not.toBeRdfIsomorphic([
      quad(
        namedNode('s1'),
        namedNode('p1'),
        namedNode('o1'),
        namedNode('g1'),
      ),
      quad(
        namedNode('s1'),
        namedNode('p1'),
        namedNode('o1'),
        namedNode('g1'),
      ),
      quad(
        namedNode('s1'),
        namedNode('p1'),
        namedNode('o1'),
        namedNode('g1'),
      ),
    ]);
  });

  it('should fail for quad arrays with equal length but different contents', () => {
    return expect(() => expect([
      quad(
        namedNode('s1'),
        namedNode('p1'),
        namedNode('o1'),
        namedNode('g1'),
      ),
      quad(
        namedNode('s2'),
        namedNode('p2'),
        namedNode('o2'),
        namedNode('g2'),
      ),
      quad(
        namedNode('s3'),
        namedNode('p3'),
        namedNode('o3'),
        namedNode('g3'),
      ),
    ]).toBeRdfIsomorphic([
      quad(
        namedNode('s1'),
        namedNode('p1'),
        namedNode('o1'),
        namedNode('g1'),
      ),
      quad(
        namedNode('s1'),
        namedNode('p1'),
        namedNode('o1'),
        namedNode('g1'),
      ),
      quad(
        namedNode('s1'),
        namedNode('p1'),
        namedNode('o1'),
        namedNode('g1'),
      ),
    ])).toThrowErrorMatchingSnapshot();
  });
});
