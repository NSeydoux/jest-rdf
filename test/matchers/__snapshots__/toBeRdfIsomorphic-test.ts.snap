// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#toBeRdfIsomorphic should fail for quad arrays with different length 1`] = `"expected [ {\\"subject\\":\\"s1\\",\\"predicate\\":\\"p1\\",\\"object\\":\\"o1\\",\\"graph\\":\\"g1\\"}, {\\"subject\\":\\"s2\\",\\"predicate\\":\\"p2\\",\\"object\\":\\"o2\\",\\"graph\\":\\"g2\\"}, {\\"subject\\":\\"s3\\",\\"predicate\\":\\"p3\\",\\"object\\":\\"o3\\",\\"graph\\":\\"g3\\"} ] to be isomorphic with [ {\\"subject\\":\\"s1\\",\\"predicate\\":\\"p1\\",\\"object\\":\\"o1\\",\\"graph\\":\\"g1\\"} ]"`;

exports[`#toBeRdfIsomorphic should fail for quad arrays with equal length but different contents 1`] = `"expected [ {\\"subject\\":\\"s1\\",\\"predicate\\":\\"p1\\",\\"object\\":\\"o1\\",\\"graph\\":\\"g1\\"}, {\\"subject\\":\\"s2\\",\\"predicate\\":\\"p2\\",\\"object\\":\\"o2\\",\\"graph\\":\\"g2\\"}, {\\"subject\\":\\"s3\\",\\"predicate\\":\\"p3\\",\\"object\\":\\"o3\\",\\"graph\\":\\"g3\\"} ] to be isomorphic with [ {\\"subject\\":\\"s1\\",\\"predicate\\":\\"p1\\",\\"object\\":\\"o1\\",\\"graph\\":\\"g1\\"}, {\\"subject\\":\\"s1\\",\\"predicate\\":\\"p1\\",\\"object\\":\\"o1\\",\\"graph\\":\\"g1\\"}, {\\"subject\\":\\"s1\\",\\"predicate\\":\\"p1\\",\\"object\\":\\"o1\\",\\"graph\\":\\"g1\\"} ]"`;

exports[`#toBeRdfIsomorphic should not fail for equal quad arrays 1`] = `"expected [ {\\"subject\\":\\"s1\\",\\"predicate\\":\\"p1\\",\\"object\\":\\"o1\\",\\"graph\\":\\"g1\\"}, {\\"subject\\":\\"s2\\",\\"predicate\\":\\"p2\\",\\"object\\":\\"o2\\",\\"graph\\":\\"g2\\"}, {\\"subject\\":\\"s3\\",\\"predicate\\":\\"p3\\",\\"object\\":\\"o3\\",\\"graph\\":\\"g3\\"} ] not to be isomorphic with [ {\\"subject\\":\\"s1\\",\\"predicate\\":\\"p1\\",\\"object\\":\\"o1\\",\\"graph\\":\\"g1\\"}, {\\"subject\\":\\"s2\\",\\"predicate\\":\\"p2\\",\\"object\\":\\"o2\\",\\"graph\\":\\"g2\\"}, {\\"subject\\":\\"s3\\",\\"predicate\\":\\"p3\\",\\"object\\":\\"o3\\",\\"graph\\":\\"g3\\"} ]"`;
